sled-agent/src/bootstrap/hardware.rs

```diff
@@ -73,6 +78,12 @@ impl HardwareMonitorWorker {
                                     warn!(self.log, "Failed to deactivate switch: {e}");
                                 }
                             }
+                            sled_hardware::HardwareUpdate::DiskAdded(disk) => {
+                                self.storage.upsert_disk(disk).await;
+                            }
+                            sled_hardware::HardwareUpdate::DiskRemoved(disk) => {
+                                self.storage.delete_disk(disk).await;
+                            }
```

Comment:
The bootstrap agent now uses a `StorageManager`, so it can create the necessary datasets on the M.2s before the sled agents launch. This will be *especially* important as we move the sled agent's configuration data out of `/var/oxide` -- these datasets must exist before the sled agent is created by RSS.